monai/transforms/utils_create_transform_ims.py

```diff
@@ -0,0 +1,220 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pathlib
+import tempfile
+from copy import deepcopy
+from glob import glob
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from monai.apps import download_and_extract
+from monai.transforms import (
+    AddChanneld,
+    Compose,
+    LoadImaged,
+    MapTransform,
+    RandFlip,
+    RandFlipd,
+    Randomizable,
+    Rotate90d,
+    ScaleIntensityd,
+    SpatialPadd,
+)
+from monai.utils.enums import CommonKeys
+from monai.utils.module import optional_import
+
+if TYPE_CHECKING:
+    import matplotlib.pyplot as plt
+
+    has_matplotlib = True
+
+else:
+    plt, has_matplotlib = optional_import("matplotlib.pyplot")
+
+
+KEYS = [CommonKeys.IMAGE, CommonKeys.LABEL]
+
+
+def get_data():
+    """Get the example data to be used.
+
+    Use MarsAtlas as it only contains 1 image for quick download and
+    that image is parcellated.
+    """
+    cache_dir = os.environ.get("MONAI_DATA_DIRECTORY") or tempfile.mkdtemp()
+    fname = "MarsAtlas-MNI-Colin27.zip"
+    url = "https://www.dropbox.com/s/ndz8qtqblkciole/" + fname + "?dl=1"
+    out_path = os.path.join(cache_dir, "MarsAtlas-MNI-Colin27")
+    zip_path = os.path.join(cache_dir, fname)
+
+    download_and_extract(url, zip_path, out_path)
+
+    image, label = sorted(glob(os.path.join(out_path, "*.nii")))
+
+    data = {CommonKeys.IMAGE: image, CommonKeys.LABEL: label}
+
```

Comment:
You could have `max_size = max(image.shape)` here and then have the `SpatialPadd` in your transform sequence to eliminate lines 76-79 and replace line 75 with `return transforms(data)`.

Reply:
But I won't know what the max size is until I've loaded the image. Which implies two sets of transforms, right?

Reply:
Yes true, typically we'd choose a fixed image size for SpatialPadd to avoid this problem.